### ISSUE TITLE
Error when attempting to use a js component which requires another local component

### ISSUE THREAD

**Issue by armincerf** [thheller/shadow-cljs#894]
I'm not sure if this is a bug/limitation of shadow-cljs or if I'm using the wrong syntax or babel configuration as I'm a beginner to the js/react world.

I've made a [minimal reproducible repo here ](https://github.com/armincerf/clojurescript-serverless-framework/tree/build-failure-requiring-js-file)

Basically, I have a js component, Header, that requires Button.js

``` Button.js
;; Button.js
import React from 'react';

export const Button = () => {
  return (
    <button>hello</button>
  );
};
```
``` Header.js
;; Header.js
import React from 'react';

import { Button } from './Button';

export const Header = () => (
  <header>
      <Button />
  </header>
);
```

This gets compiled by babel as per this part of the guide https://shadow-cljs.github.io/docs/UsersGuide.html#_javascript_dialects

The outputted gen/Header.js file looks like this:

``` js
import React from 'react';
import { Button } from './Button';
export const Header = () => /*#__PURE__*/React.createElement("header", null, /*#__PURE__*/React.createElement(Button, null));
```

However, trying to use this header component from cljs throws the following error:
```
Header.js:3 Uncaught ReferenceError: Button$$module$gen$Button is not defined
    at Header$$module$gen$Header (Header.js:3)
    at renderWithHooks (react-dom.development.js:695)
    at beginWork (react-dom.development.js:900)
    at HTMLUnknownElement.callCallback (react-dom.development.js:1408)
    at Object.invokeGuardedCallbackImpl (react-dom.development.js:1414)
    at invokeGuardedCallback (react-dom.development.js:10)
    at beginWork$1 (react-dom.development.js:1713)
    at performUnitOfWork (react-dom.development.js:1251)
    at workLoopSync (react-dom.development.js:1252)
    at performSyncWorkOnRoot (react-dom.development.js:1186)
Header$$module$gen$Header @ Header.js:3
renderWithHooks @ react-dom.development.js:695
beginWork @ react-dom.development.js:900
callCallback @ react-dom.development.js:1408
invokeGuardedCallbackImpl @ react-dom.development.js:1414
invokeGuardedCallback @ react-dom.development.js:10
beginWork$1 @ react-dom.development.js:1713
performUnitOfWork @ react-dom.development.js:1251
workLoopSync @ react-dom.development.js:1252
performSyncWorkOnRoot @ react-dom.development.js:1186
scheduleWork @ react-dom.development.js:1744
updateContainer @ react-dom.development.js:1380
eval @ react-dom.development.js:1356
unbatchedUpdates @ react-dom.development.js:1251
legacyRenderSubtreeIntoContainer @ react-dom.development.js:1356
exports.render @ react-dom.development.js:1767
todo_mvc$core$start @ core.cljs:9
(anonymous) @ (index):12
react-dom.development.js:986 The above error occurred in the <Header$$module$gen$Header> component:
    in Header$$module$gen$Header
```

The cljs in question (note that importing and rendering "../gen/Button.js" works as expected)
```
(ns todo-mvc.core
  (:require
   ["../gen/Header.js" :refer (Header)]
   ["react" :as r]
   ["react-dom" :as rdom]))

(defn ^:export start
  []
  (rdom/render (r/createElement Header) (js/document.getElementById "app")))
```

Is this something I can work around? I imagine other people must be using javascript components in cljs projects and it's hard to believe no one is importing other local files so there must be something I'm doing wrong here, but I've spent quite a lot of time trying different solutions with no luck 🤷

**Comment by thheller** [comment]
I have identified the cause but not sure what I can do about it.

I recently bumped the default `:output-feature-set` to `:es6`, which now causes the Closure Compiler to rewrite the code to use `const`.

```jsx
export const Button = () => {
  return (
    <button>hello</button>
  );
};

// becomes

const Button$$module$gen$Button = function() {
  return module$node_modules$react$index.createElement("button", null, "hello");
};
```

whereas with the previous default it was `var Button$$module$gen$Button`. Given how `var` works that worked just fin. The `const` however has stricter scoping rules so the other file, running in a separate `eval` cannot "see" it properly.

There are two options to work arround this that I can think of:

- `:compiler-options {:output-feature-set :es5}`, so reverting back to the previous default. That'll make it use `var` again and everything just works, at the expense of being ES5 again. Not ideal but maybe not critical.
- `:devtools {:loader-mode :script}` that'll disable the default `eval` loader and instead use `script` tags. Here `const` appears to work fine but this probably only works for the initial page load. I suspect that this will break when loading files via the REPL or hot-reload.

I might need to do a deep dive into some of the closure rewriting code and see if I can find some way to make this more reliable.
